### ISSUE TITLE
QueryWithVars using `after:` fails

### ISSUE THREAD

**Issue by llonchj** [dgraph-io/dgraph#2099]
If you suspect this could be a bug, follow the template.

- What version of Dgraph are you using?

- Have you tried reproducing the issue with latest release?

- What is the hardware spec (RAM, OS)?

MacOS 10.13.3 16Gb

- Steps to reproduce the issue (command/config used to run Dgraph).

This is what i have

```{
    items(func:has(<type:EmailAddress>), orderasc:<type:EmailAddress>, after:$i) @filter(regexp(<address>, /$var/)) {
        uid
        key: <type:EmailAddress>
    }
}```

and vars

```map[$var:@example\.com $i:0x1010]```

- Expected behaviour and actual result.

```rpc error: code = Unknown desc = Expecting a comma. Got: lex.Item [9] "i"```

**Comment by sboorlagadda** [comment]
It looks like GraphQL & Dgraph vars are not supported for `after`. If we set the actual value it works.
```
query q($var: string) {

	items(func:has(<type:EmailAddress>), orderasc:<type:EmailAddress>, after: 0x1010) @filter(regexp(<type:EmailAddress>, /$var/)) {
                uid
        	key: <type:EmailAddress>
    	}
}
``` 
But none of these below seem to work.

*Dgraph Var*
```
query q($var: string, $i: string) {
	var (func: uid($i)) {
		a as uid
	}
	items(func:has(<type:EmailAddress>), orderasc:<type:EmailAddress>, after: a) @filter(regexp(<type:EmailAddress>, /$var/)) {
        	uid
        	key: <type:EmailAddress>
    	}
}
```
*GraphQL Var*
```
query q($i: string) {
	me(func: has(friend), after: $i) {
		uid
		name
		friend {
			count(uid)
		}
	}
}
```

**Comment by pawanrawal** [comment]
Dgraph query uid/value variables don't make sense as being used for `after`. We should support GraphQL variables for `after` though.

**Comment by llonchj** [comment]
@sboorlagadda I agree, `after` and likely the other variables involved in a `q(...)` should be supported via graphql variables to avoid potential query injection vulnerabilities.

**Comment by pawanrawal** [comment]
Merged to master with e05f0fd6ef685773bf2424dbcfed4d810accf0bf.

**Comment by llonchj** [comment]
Thanks! Will test and give feedback
